Modules/_sqlite/cursor.c

```diff
@@ -512,6 +512,7 @@ _pysqlite_query_execute(pysqlite_Cursor* self, int multiple, PyObject* operation
             goto error;
         }
         rc = pysqlite_statement_create(self->statement, self->connection, operation);
```

Comment:
Tracking should be done on the constructor, not on the code calling it, as is not it's responsability. This way there is no need for synchronization and eliminates the possibility of getting a badly crated object.

Reply:
Yes, that's better. I'll move the allocation there as well, for consistency.

Reply:
I've moved allocation and GC tracking over to `pysqlite_statement_create`. PTAL.